platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformService.java

```diff
@@ -57,6 +57,11 @@ public PlatformService(ConfigClient configClient) {
     this.serviceName = appConfig.getString(SERVICE_NAME_CONFIG);
   }
 
+  public PlatformService(Config appConfig){
```

Comment:
I don't know the context of this change, but I think this is likely to produce bugs when used - it would leave the configClient field null, and I know this is used by many subclients. Rather than changing the framework, consider creating an implementation of configclient for your use case that always returns the same config.

Reply:
Basically this is the structure that I'm testing:

/app/resources/configs/consolidated-service/application.conf
/app/resources/configs/serviceA/application.conf
/app/resources/configs/serviceA/overrides/application.conf
/app/resources/configs/serviceB/application.conf
/app/resources/configs/serviceB/overrides/application.conf
/app/resources/configs/serviceC/application.conf
/app/resources/configs/serviceC/overrides/application.conf

I can create a config client like this - ConfigClientFactory.getClient(String serviceName, String ...overrides) which will return DirectoryBasedConfigClientWithOverrides instance.


`public class DirectoryBasedConfigClientWithOverrides extends DirectoryBasedConfigClient {

  private final String serviceName;
  private final String[] overrides;

  /**
   * @param baseDir Ex: /app/resources/configs
   * @param serviceName - name of the service this config belongs to
   * @param overrides - Specify overrides from coarsest to fine grained scope. Ex: cluster, pod, container. T
   * hen the config will be loaded from fine grained scope and
   * fall back to more coraser grain scope in reverse order.
   * Ex: container config => fallback on pod config => fall back on cluster config => fallback on service config
   */
  public DirectoryBasedConfigClientWithOverrides(String baseDir, String serviceName,
      String... overrides) {
    super(baseDir);
    this.overrides = overrides;
    this.serviceName = serviceName;
  }

  @Override
  public Config getConfig() {
    String pathFormat = "%s/%s";
    String path = String.format(pathFormat, baseDir, serviceName);
    Config config = getConfigFromPath(path)
        .withFallback(getConfigFromPath(String.format(pathFormat, baseDir, COMMON_DIRECTORY)));

    List<Config> overrideConfigs = new ArrayList<>();
    if (overrides != null) {
      List<String> commonDirArgs = new ArrayList<>();
      commonDirArgs.add(baseDir);
      commonDirArgs.add(COMMON_DIRECTORY);
      List<String> args = new ArrayList<>();
      args.add(baseDir);
      args.add(serviceName);
      for (String override : overrides) {
        pathFormat = pathFormat + "/%s";
        args.add(override);
        commonDirArgs.add(override);
        path = String.format(pathFormat, args.toArray());
        overrideConfigs.add(getConfigFromPath(path)
            .withFallback(getConfigFromPath(String.format(pathFormat, commonDirArgs.toArray()))));
      }

      if (!overrideConfigs.isEmpty()) {
        Config result = overrideConfigs.get(overrideConfigs.size() - 1);
        for (int i = overrideConfigs.size() - 2; i > 0; i--) {
          result = result.withFallback(overrideConfigs.get(i));
        }
        config = result.withFallback(config);
      }
    }
    return config;
  }
}
`

Reply:
> Basically this is the structure that I'm testing:

This aligns with what I did with grpc, so I'm a bit confused. I didn't change any of the config classes for it - I just passed the override path in to the existing constructor: 
https://github.com/hypertrace/service-framework/blob/main/platform-grpc-service-framework/src/main/java/org/hypertrace/core/serviceframework/grpc/ConsolidatedGrpcServiceContainerEnvironment.java#L29

Reply:
Discussed offline - the tl;dr is - use the existing config resolution (to avoid all the custom resolution added above), then put the result in a dummy client that just echoes back the config it was given at construction time. In essence, we're using config client as a wrapper to do resolution outside rather than inside the service.